src/nodes.coffee

```diff
@@ -4809,6 +4813,36 @@ exports.If = class If extends Base
   unfoldSoak: ->
     @soak and this
 
+  processedCondition: ->
+    @processedConditionCache ?= if @type is 'unless' then @condition.invert() else @condition
+
+  isStatementAst: (o) ->
+    o.level is LEVEL_TOP
```

Comment:
This is what needed `o.level` to be set - in order to more closely mimic Babel AST, `If` ASTs should be of type `IfStatement` (ie a JS `if` statement) if we're at `LEVEL_TOP` and a `ConditionalExpression` (ie a JS ternary) otherwise